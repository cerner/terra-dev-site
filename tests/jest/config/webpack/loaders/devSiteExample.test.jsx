import 'regenerator-runtime/runtime';

const path = require('path');
const { runLoaders } = require('loader-runner');

describe('devSiteExample file test', () => {
  it('when a valid css file is present', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendExample.jsx'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSiteExample'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when an unresolvable css file is present', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendExample2.jsx'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSiteExample'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });

  it('when no css file is present', (done) => {
    runLoaders({
      resource: path.resolve(__dirname, './pretendExample3.jsx'),
      loaders: [
        path.resolve(__dirname, '../../../../../config/webpack/loaders/devSiteExample'),
      ],
    }, (err, result) => {
      if (err) return done(err);
      expect(result.result[0]).toMatchSnapshot();
      return done();
    });
  });
});

