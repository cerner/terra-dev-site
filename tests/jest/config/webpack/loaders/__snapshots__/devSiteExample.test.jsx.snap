// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`devSiteExample file test when a valid css file is present 1`] = `
"import React from 'react';
import Example from '/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExample.jsx';
import Code from '/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExample.jsx?dev-site-codeblock';
import ExampleTemplate from 'terra-dev-site/lib/loader-components/_ExampleTemplate';
export default ({ title, description, isExpanded }) => (
        <ExampleTemplate
          title={ title || 'Pretend Example'}
          description={description}
          example={<Example />}
          exampleSrc={<Code />}
          isExpanded={isExpanded}
        />
      );"
`;

exports[`devSiteExample file test when an unresolvable css file is present 1`] = `
"import React from 'react';
import Example from '/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExample2.jsx';
import Code from '/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExample2.jsx?dev-site-codeblock';
import ExampleTemplate from 'terra-dev-site/lib/loader-components/_ExampleTemplate';
export default ({ title, description, isExpanded }) => (
        <ExampleTemplate
          title={ title || 'Pretend Example 2'}
          description={description}
          example={<Example />}
          exampleSrc={<Code />}
          isExpanded={isExpanded}
        />
      );"
`;

exports[`devSiteExample file test when no css file is present 1`] = `
"import React from 'react';
import Example from '/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExample3.jsx';
import Code from '/Users/dm068655/Documents/ExternalR/terra-dev-site/tests/jest/config/webpack/loaders/pretendExample3.jsx?dev-site-codeblock';
import ExampleTemplate from 'terra-dev-site/lib/loader-components/_ExampleTemplate';
export default ({ title, description, isExpanded }) => (
      <ExampleTemplate
        title={ title || 'Pretend Example 3'}
        description={description}
        example={<Example />}
        exampleSrc={<Code />}
        isExpanded={isExpanded}
      />
    );"
`;
